wikimedia/wikipedia-ios
# [Widgets] Update Top read widget to use new data fetching pattern

**Phabricator:** https://phabricator.wikimedia.org/T327556

### Notes
This PR updates the top read widget to use the new data models from https://github.com/wikimedia/wikipedia-ios/pull/4502 and a non-Core Data independent fetching pattern.

### Test Steps

We should rely on QA to exhaustively test this. Note that depending on the number of widgets you've added to your home screen and have been cached, you may run into this bug https://phabricator.wikimedia.org/T334771.

- Remove all existing Wikipedia home screen widgets from your device
- Try adding a Top read widget of each size to your device, confirm it renders
- If it doesn't render, you can run the `WidgetController`'s associated `func fetch...` from the main app to confirm if it's a Codable/API issue or the image memory limit bug above.
